doc/source/whatsnew/v0.17.0.txt

```diff
@@ -204,6 +204,53 @@ The support math functions are `sin`, `cos`, `exp`, `log`, `expm1`, `log1p`,
 
 These functions map to the intrinsics for the NumExpr engine.  For Python
 engine, they are mapped to NumPy calls.
+=======
+
+Changes to Excel with ``MultiIndex``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+In version 0.16.2 a ``DataFrame`` with ``MultiIndex`` columns could not be written to Excel via ``to_excel``.
+That functionality has been added (:issue:`10564`), along with updating  ``read_excel`` so that the data can
+be read back with no loss of information by specifying which columns/rows make up the ``MultiIndex``
+in the `header` and `index_col` parameters (:issue:`4679`)
+
+.. ipython:: python
+
+    df = pd.DataFrame([[1,2,3,4], [5,6,7,8]],
+                      columns = pd.MultiIndex.from_product([['foo','bar'],['a','b']],
+                                                           names = ['col1', 'col2']),
+                      index = pd.MultiIndex.from_product([['j'], ['l', 'k']],
+                                                         names = ['i1', 'i2']))
+
+    df
+    df.to_excel('test.xlsx')
+
+    df = pd.read_excel('test.xlsx', header=[0,1], index_col=[0,1])
+    df
+    
+.. ipython:: python
+   :suppress:
+
+   import os
+   os.remove('test.xlsx')
+
+Previously, it was necessary to specifying the ``has_index_names`` argument in ``read_excel``
+if the serialized data had index names.  For version 0.17 the ouptput format of ``to_excel``
+has been changed to make this keyword unnecessary - the change is shown below.
+
+**Old**
+
+.. image:: _static/old-excel-index.png
+
+**New**
+
+.. image:: _static/new-excel-index.png
+
+.. warning::
+
+    Excel files saved in version 0.16.2 or prior that had index names will still able to be read in,
```

Comment:
There is a back compat test [here](https://github.com/pydata/pandas/pull/10967/files#diff-8db63cd3c37f8795b0b65cf5eee35946R603).  

It's annoying, but because the old format differed on whether there were names or not, there's no way I can see to just do the right thing (same ambiguity as above around a row of blank data).  The reason I didn't change the default it to True is to preserve backwards compat  - any old code should just work.  Changing the default could introduce a pretty subtle bug in the corner case (if a user's data first row was all blanks it would be lost).